src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -2213,18 +2213,613 @@ class C
+                Diagnostic(ErrorCode.ERR_PartialPropertyMissingImplementation, "this").WithArguments("C2.this[ref readonly int]").WithLocation(9, 24),
+                // (10,24): error CS9301: Partial property 'C2.this[in int]' must have a definition part.
+                //     public partial int this[in int i] { get => i; set { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyMissingDefinition, "this").WithArguments("C2.this[in int]").WithLocation(10, 24),
+                // (10,24): error CS0111: Type 'C2' already defines a member called 'this' with the same parameter types
+                //     public partial int this[in int i] { get => i; set { } }
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "this").WithArguments("this", "C2").WithLocation(10, 24));
+        }
+
+        [Fact]
+        public void TypeDifference_IndexerParameter()
+        {
+            var source = """
+                partial class C
+                {
+                    public partial int this[int i] { get; set; }
+                    public partial int this[string s] { get => 1; set { } }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (3,24): error CS9300: Partial property 'C.this[int]' must have an implementation part.
+                //     public partial int this[int i] { get; set; }
+                Diagnostic(ErrorCode.ERR_PartialPropertyMissingImplementation, "this").WithArguments("C.this[int]").WithLocation(3, 24),
+                // (4,24): error CS9301: Partial property 'C.this[string]' must have a definition part.
+                //     public partial int this[string s] { get => 1; set { } }
+                Diagnostic(ErrorCode.ERR_PartialPropertyMissingDefinition, "this").WithArguments("C.this[string]").WithLocation(4, 24));
+        }
+
+        [Fact]
+        public void NullabilityDifference_IndexerParameter()
+        {
+            var source = $$"""
+                #nullable enable
+                partial class C
+                {
+                    public partial int this[string s] { get; set; }
+                    public partial int this[string? s] { get => 1; set { } }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (5,24): warning CS9308: Partial property declarations 'int C.this[string s]' and 'int C.this[string? s]' have signature differences.
+                //     public partial int this[string? s] { get => 1; set { } }
+                Diagnostic(ErrorCode.WRN_PartialPropertySignatureDifference, "this").WithArguments("int C.this[string s]", "int C.this[string? s]").WithLocation(5, 24));
+        }
+
+        [Fact]
+        public void DynamicDifference_IndexerParameter()
+        {
+            var source = """
+                #nullable enable
```

Comment:
nit: nullable is not necessary here

Reply:
I'll try to avoid in future tests, but don't want to clean this up for existing tests, as it requires either adjusting the baselines or leaving in extra blank lines in the test source.